firedantic/tests/tests_async/test_model.py

```diff
@@ -534,4 +536,116 @@ async def test_save_with_exclude_unset(configure_db) -> None:
     document = await Profile._get_col_ref().document(document_id).get()
 
     data = document.to_dict()
-    assert data == {"name": "", "photo_url": None}
+    assert data == {"name": "", "email": None, "photo_url": None}
+
+
+@pytest.mark.asyncio
+async def test_create_in_transaction(configure_db) -> None:
+    """Test creating a model in a transaction. Test case from README."""
+
+    @async_transactional  # type: ignore
+    async def create_in_transaction(transaction, email) -> Profile:
+        """Creates a Profile in a transaction"""
+        try:
+            await Profile.find_one({"email": email}, transaction=transaction)
+            raise ValueError(f"Profile already exists with email: {email}")
+        except ModelNotFoundError:
+            p = Profile(email=email)
+            await p.save(transaction=transaction)
+            return p
+
+    transaction = CONFIGURATIONS["db"].transaction()
+    p = await create_in_transaction(transaction, "test@example.com")
+    assert isinstance(p, Profile)
+    assert p.id
+
+    transaction2 = CONFIGURATIONS["db"].transaction()
+    with pytest.raises(ValueError) as excinfo:
+        await create_in_transaction(transaction2, "test@example.com")
+    assert str(excinfo.value) == "Profile already exists with email: test@example.com"
+
+
+@pytest.mark.asyncio
+async def test_delete_in_transaction(configure_db) -> None:
+    """Test deleting a model in a transaction."""
+
+    @async_transactional  # type: ignore
+    async def delete_in_transaction(
+        transaction: AsyncTransaction, profile_id: str
+    ) -> dict:
+        """Deletes a Profile in a transaction."""
+        try:
+            p = await Profile.get_by_id(profile_id, transaction=transaction)
+            await p.delete(transaction=transaction)
+            return {"id": profile_id}
+        except ModelNotFoundError:
+            raise ValueError(f"Profile not found: {profile_id}")
+
+    p = Profile(name="Foo")
+    await p.save()
+    assert p.id
+
+    transaction = CONFIGURATIONS["db"].transaction()
+    result = await delete_in_transaction(transaction, p.id)
+    assert isinstance(result, dict)
+    assert result == {"id": p.id}
```

Comment:
You could simplify the test to just return one value (the id) and not need to check it's a dictionary and make a matching dictionary to compare it with.

Reply:
I've taken your suggestion and modified the function to return the profile_id that was passed into the function.